src/himali/Assignment12.java

```diff
@@ -0,0 +1,85 @@
+package himali;
+
+public class Assignment12 {
+	int i;
+	int minimumNumber(int[] num) {
+		int min=num[0];
+		for(i=0;i<num.length;i++) {
+			if(num[i]<min) {
+				min=num[i];
+			}
+		}
+		return min;
+		}
+	
+	int maxNumber(int[] num) {
+		int max=num[0];
+		for(i=0;i<num.length;i++) {
+			if(num[i]>max) {
+				max=num[i];
+			}
+			
+		}
+		return max;
+		
+		
+	}
+	double averageOfAllNumbers(int[] num) {
+		int sum = 0;
+		for(i = 0; i < num.length; i++) {
+			sum = sum + num[i];
+		}
+		double avg = sum / num.length;
+		return avg;
+		
+	}
+	int averageOfMinMaxNumber(int min,int max) {
+		int averageMinMax=0;
```

Comment:
I think we have requirement where you need to find max & min number then you need to find avg of both, so this expecting two variable in method looks wrong as per the implementation perspective, user will give us array, not min & max value. I hope you got my point.


You can write code like this, I remove variable and directly doing sum of min & max, no problem, if you are not comfortable to write code like below, just optimize.

int avgOfMinMax = (minNumber(input)+maxNumber(input))/2